docs/Hooks.md

```diff
@@ -104,17 +104,16 @@ fastify.addHook('preHandler', (request, reply, next) => {
 
 Note that in the `'preHandler'` and `'onSend'` hook the request and reply objects are different from `'onRequest'`, because the two arguments are [`request`](https://github.com/fastify/fastify/blob/master/docs/Request.md) and [`reply`](https://github.com/fastify/fastify/blob/master/docs/Reply.md) core Fastify objects.
 
-If you are using the `onSend` hook you can update the payload, but not overwrite it, for example:
+If you are using the `onSend` hook you can update or overwrite the payload:
+The next hook in the order of execution will get the updated/overwritten payload.
 ```js
-// this is valid
-fastify.addHook('onSend', (request, reply, payload, next) => {
```

Comment:
This would be the same why we should introduce a seperate `ctx` parameter ?

```js
fastify.addHook('onSend', (request, reply, next) => {
  reply.payload.hello = 'world'
})
```

Reply:
Yeah, that is another solution, probably even better.
@fastify/fastify do we all agree?

Reply:
👍

Reply:
👍

Reply:
Yeah 👍 
And `ctx` (context I think) is already used for `reply.context`(https://github.com/fastify/fastify/blob/master/lib/reply.js#L15)